static/js/lib/learning_resources.js

```diff
@@ -27,24 +34,131 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const parseDateFilter = (filter: string) => {
+  const format = /(now)(\+)?(\d+)?([Md])?/
+  const match = format.exec(filter)
+  if (match) {
+    let dt = moment()
+    if (match[3] && match[4]) {
+      dt = dt.add(match[3], match[4] === "d" ? "days" : "months")
+    }
+    return dt
+  }
+}
+
+export const inDateRanges = (run: CourseRun, availabilities: Array<string>) => {
+  let inRange = false
+  availabilities.forEach(availability => {
+    if (AVAILABILITY_MAPPING[availability]) {
+      const from = parseDateFilter(
+        AVAILABILITY_MAPPING[availability].filter.from
+      )
+      const to = parseDateFilter(AVAILABILITY_MAPPING[availability].filter.to)
+      const startDate = run.best_start_date
+        ? moment(run.best_start_date, dateFormat)
+        : null
+      if (
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(from) ||
+          // $FlowFixMe: okay to compare moments
+          startDate >= from) &&
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(to) ||
+          // $FlowFixMe: okay to compare moments
+          startDate <= to)
+      ) {
```

Comment:
for this check here you might be able to use moment's `.isBetween` helper

Reply:
Replaced <= and >= with isSameOrBefore, isSameOrAfter. Avoided isBetween because of the possibility of null from/to values.